assets/app/scripts/directives/events.js

```diff
@@ -11,16 +11,66 @@ angular.module('openshiftConsole')
       },
       templateUrl: 'views/directives/events.html',
       controller: function($scope){
+        $scope.events = {};
+        $scope.filter = {
+          text: ''
+        };
+
+        var filterForResource = function(events) {
+          return _.filter(events, function(event) {
+            return event.involvedObject.kind === $scope.resourceKind &&
+                   event.involvedObject.name === $scope.resourceName;
+          });
+        };
+
+        var sortedEvents = [];
+        var sortEvents = function() {
+          // TODO: currentField is renamed in angular-patternfly 3.0
+          var sortID = _.get($scope, 'sortConfig.currentField.id', 'lastTimestamp'),
+              order = $scope.sortConfig.isAscending ? 'asc' : 'desc';
+          sortedEvents = _.sortByOrder($scope.events, [sortID], [order]);
+        };
+
+        var applyFilter = $filter('filter');
+        var filterForKeyword = function() {
+          $scope.filteredEvents = applyFilter(sortedEvents, $scope.filter.text);
```

Comment:
throttle here as well? if I type 10 letters in a second, filtering 4 times should be plenty